src/main/java/org/opensearch/ad/model/AnomalyDetector.java

```diff
@@ -211,8 +212,13 @@ public AnomalyDetector(
         if (detectionInterval == null) {
             throw new IllegalArgumentException("Detection interval should be set");
         }
-        if (shingleSize != null && shingleSize < 1) {
-            throw new IllegalArgumentException("Shingle size must be a positive integer");
+        if (shingleSize != null && (shingleSize < 1 || shingleSize > AnomalyDetectorSettings.MAX_SHINGLE_SIZE)) {
```

Comment:
Thanks for adding this check. How about build a util method so historical analysis can reuse it?

Reply:
sure, added

Reply:
Not sure if I missed something, but seems the util method not added yet?

Reply:
not yet.  Will send the new commit after retesting.